ArduCopter/AP_Arming_Copter.cpp

```diff
@@ -138,8 +138,10 @@ bool AP_Arming_Copter::barometer_checks(bool display_failure)
         // that may differ from the baro height due to baro drift.
         const auto &ahrs = AP::ahrs();
         const bool using_baro_ref = !ahrs.has_status(AP_AHRS::Status::PRED_HORIZ_POS_REL) && ahrs.has_status(AP_AHRS::Status::PRED_HORIZ_POS_ABS);
+        float pos_d_m = 0;
+        UNUSED_RESULT(AP::ahrs().get_relative_position_D_origin_float(pos_d_m));
         if (using_baro_ref) {
-            if (fabsf(copter.inertial_nav.get_position_z_up_cm() - copter.baro_alt) > PREARM_MAX_ALT_DISPARITY_CM) {
+            if (fabsf(pos_d_m) > PREARM_MAX_ALT_DISPARITY_CM) {
```

Comment:
Unit failure - m vs cm

Reply:
... also, this is not the same check - you're asking the  AHRS for an altitude, not the baro library now, where the comment is definitely wanting you to be looking at the baro...